subsys/bluetooth/services/fast_pair/fp_keys.c

```diff
@@ -533,5 +534,37 @@ static int fp_keys_uninit(void)
 	return 0;
 }
 
+int fp_keys_bond_save(const struct bt_conn *conn)
+{
+	if (!IS_ENABLED(CONFIG_BT_FAST_PAIR_BOND_MANAGER)) {
+		__ASSERT_NO_MSG(false);
+		return -ENOTSUP;
+	}
+	__ASSERT_NO_MSG(bt_fast_pair_is_ready());
+
+	struct fp_procedure *proc = &fp_procedures[bt_conn_index(conn)];
+	const struct fp_account_key *account_key;
+	struct bt_conn_info conn_info;
+	int err;
+
+	err = bt_conn_get_info(conn, &conn_info);
+	if (err) {
+		LOG_ERR("Failed to get local conn info (err %d)", err);
+		return err;
+	}
+
+	__ASSERT_NO_MSG(is_key_generated(proc->state));
+
+	if (proc->state == FP_STATE_USE_TEMP_KEY) {
+		/* Initial pairing - the Account Key is not written yet. */
+		account_key = NULL;
+	} else {
+		/* Subsequent pairing - the Account Key is already known. */
+		account_key = (const struct fp_account_key *)proc->aes_key;
```

Comment:
Then we would have to cast it to be able to use `fp_crypto_aes128_ecb_encrypt` and similar crypto functions. Either way we will need to cast something. I'd keep it as is to have casting in only one place. Also having `aes_key` type as `fp_account_key` may be confusing. In Initial Pairing procedure it's not an account key.

Reply:
I am just a bit worried that this cast could go wrong if we append an additional data field to `fp_account_key` structure at some point. That's why I suggested the change.

Alternatively, maybe we could create the struct object here locally and then pass it to function to avoid the risky cast?

Reply:
Using a structure here could also help us if we opt for using opaque key identifiers instead of byte arrays here (PSA API)